src/modules/sensors/sensors.cpp

```diff
@@ -447,6 +447,10 @@ Sensors::adc_poll()
 		int   bat_voltage_v_chan[BOARD_NUMBER_BRICKS] = BOARD_BATT_V_LIST;
 		int   bat_voltage_i_chan[BOARD_NUMBER_BRICKS] = BOARD_BATT_I_LIST;
 
+		if (_parameters.battery_adc_channel >= 0) {  // overwrite default
```

Comment:
We need a better battery/power architecture, but at the moment the "right" way to do this would be setting the ADC channel in the board config.

Reply:
In posix-configs/bbblue/px4.config, I have the following comments that could explain why this option was added. Fixing the setting in board config seems could not satisfy the need to specify the main power source for the case of BeagleBone Blue.

# Three possible main power battery sources for BBBlue:
# 1. onboard 2S LiPo battery connector, which is connect to ADC channel 6
# 2. onboard 9-18V DC Jack, which is connect to ADC channel 5. This is the board default.
# 3. other power source (e.g., LiPo battery more than 4S/18V). 
#    Scale the voltage to below 1.8V and connect it to ADC channel # 0, 1, 2 or 3. 
param set BAT_ADC_CHANNEL 	5